zubhub_frontend/zubhub/src/views/create_activity/createActivityScripts.js

```diff
@@ -0,0 +1,474 @@
+            );
+
+            if (error) console.log('initUploadError', error);
+          });
+      }
+    });
+  }
+};
+
+export const uploadImage = (image, state, props, handleSetState, label) => {
+  const args = {
+    t: props.t,
+    token: props.auth.token,
+  };
+
+  return API.shouldUploadToLocal(args).then(res => {
+    if (res && res.local === true) {
+      return uploadImageToLocal(image, state, props, handleSetState);
+    } else if (res && res.local === false) {
+      return uploadImageToDO(image, state, props, handleSetState, label);
+    }
+  });
+};
+
+export const uploadImageToLocal = (image, state, props, handleSetState) => {
+  let url =
+    process.env.REACT_APP_NODE_ENV === 'production'
+      ? process.env.REACT_APP_BACKEND_PRODUCTION_URL + '/api/'
+      : process.env.REACT_APP_BACKEND_DEVELOPMENT_URL + '/api/';
+  url = url + 'upload-file-to-local/';
+
+  const formData = new FormData();
+  formData.append('file', image);
+  formData.append('key', `project_images/${nanoid()}`);
+  return new Promise((resolve, reject) => {
+    const um = new UploadMedia(
+      'image',
+      url,
+      formData,
+      state,
+      props,
+      handleSetState,
+      resolve,
+      reject,
+    );
+    um.upload();
+  });
+};
+
+export const uploadImageToDO = (image, state, props, handleSetState, label) => {
+  return new Promise((resolve, reject) => {
+    const params = {
+      Bucket: `${doConfig.bucketName}`,
+      Key: `${label}/${nanoid()}`,
+      Body: image,
+      ContentType: image.type,
+      ACL: 'public-read',
+    };
+
+    DO.upload(params, err => {
+      reject(err.message);
+    })
+      .on('httpUploadProgress', e => {
+        const progress = Math.round((e.loaded * 100.0) / e.total);
+        const { media_upload } = state;
+        const upload_info = JSON.parse(
+          JSON.stringify(media_upload.upload_info),
+        );
+        upload_info[image.name] = progress;
+
+        let total = 0;
+        Object.keys(upload_info).forEach(each => {
+          total = total + upload_info[each];
+        });
+
+        total = total / Object.keys(upload_info).length;
+
+        handleSetState({
+          media_upload: {
+            ...media_upload,
+            upload_info,
+            upload_percent: total,
+          },
+        });
+      })
+      .send((err, data) => {
+        if (err) {
+          if (err.message.startsWith('Unexpected')) {
+            const error = props.t('createProject.errors.unexpected');
+            reject(error);
+          } else {
+            reject(err.message);
+          }
+        } else {
+          const secure_url = data.Location;
+          const public_id = data.Key;
+          resolve({ image_url: secure_url, public_id });
+        }
+      });
+  });
+};
+
+export class UploadMedia {
+  constructor(
+    type,
+    url,
+    formData,
+    state,
+    props,
+    handleSetState,
+    resolve,
+    reject,
+  ) {
+    this.xhr = new XMLHttpRequest();
```

Comment:
Images can be uploaded using the fetch method in a simple way can you please checkout?

Reply:
yes I see thanks for mentioning that, i was just trying to use what already works for projects, make things work then i will try to optimize those block